System.Containers.Tasks/build/System.Containers.Tasks.targets

```diff
@@ -11,38 +11,60 @@
+            <!-- TODO: Test this scenario -->
+            <ContainerEntrypoint Condition="'$(ContainerEntrypoint)' == '' and '$(UseAppHost)' != 'true'">dotnet $(TargetFileName)</ContainerEntrypoint>
+            <ContainerEntrypoint Condition="'$(ContainerEntrypoint)' == '' and '$(UseAppHost)' == 'true'">$(ContainerWorkingDirectory)/$(AssemblyName)$(_NativeExecutableExtension)</ContainerEntrypoint>
+            <!-- Could be semicolon-delimited -->
             <ContainerEntrypointArgs Condition="'$(ContainerEntrypointArgs)' == ''"></ContainerEntrypointArgs>
         </PropertyGroup>
+
+        <ItemGroup>
+            <ContainerLabel Condition="@(ContainerLabel) == ''"></ContainerLabel>
+        </ItemGroup>
+
+        <ItemGroup>
+            <!-- Storing this for later use:  or %(ContainerPort.Identity) == '80' and @(ContainerPort->WithMetadataValue('Type', 'tcp')->Count()) == 0) -->
+            <ContainerPort Include="80" Type="tcp" Condition="%(ProjectCapability.Identity) == 'AspNetCore' and
+                                                              %(ContainerPort.Identity) != '80'" />
+
+            <ContainerEnvironmentVariable Include="ASPNETCORE_URLS" Value="http://localhost:5000;https://localhost:5001"
+                                                                    Condition="%(ProjectCapability.Identity) == 'AspNetCore' and
+                                                                               %(ContainerEnvironmentVariable.Identity) != 'ASPNETCORE_URLS' "/>
+        </ItemGroup>
+
+        <ParseContainerProperties ContainerBaseImage="$(ContainerBaseImage)"
+                                  ContainerImageName="$(ContainerImageName)"
+                                  ContainerImageTag="$(ContainerImageTag)">
+
+            <Output TaskParameter="ParsedContainerRegistry" PropertyName="_ContainerRegistry" />
+            <Output TaskParameter="ParsedContainerImage" PropertyName="_ContainerImage" />
+            <Output TaskParameter="ParsedContainerTag" PropertyName="_ContainerTag" />
+            <Output TaskParameter="NewImageName" PropertyName="_NewImageName" />
+            <Output TaskParameter="NewImageTag" PropertyName="_NewImageTag" />
+        </ParseContainerProperties>
     </Target>
 
-    <Target Name="PublishContainer" DependsOnTargets="$(PublishContainerDependsOn)" BeforeTargets="Publish">
-        <CreateNewImage BaseImageName="$(ContainerBaseImageName)"
-                        BaseImageTag="$(ContainerBaseImageTag)"
-                        InputRegistryURL="$(ContainerInputRegistryURL)"
-                        OutputRegistryURL="$(ContainerOutputRegistryURL)"
+    <Target Name="PublishContainer" DependsOnTargets="$(PublishContainerDependsOn)" AfterTargets="ComputeContainerConfig" BeforeTargets="Publish">
+        <CreateNewImage BaseRegistry="$(_ContainerRegistry)"
+                        BaseImageName="$(_ContainerImage)"
+                        BaseImageTag="$(_ContainerTag)"
```

Comment:
`ParseContainerProperties` exists to parse everything and catch any potential issues with the given properties. It has to parse each piece here, and I figured we might as well have it output that data. This way CreateNewImage doesn't need to re-parse data when it calls the API.

This can change of course